src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitterUTest.java

```diff
@@ -242,13 +263,48 @@ public void setup() {
         Whitebox.setInternalState(assignmentSubmitter, "assignmentValidator", assignmentValidator);
         Whitebox.setInternalState(assignmentSubmitter, "studentRepository", studentRepository);
         Whitebox.setInternalState(assignmentSubmitter, "studentService", studentService);
+    }
 
+    @Test
+    public void submitCourseAttendancesHappyPath() throws Exception {
+        setupMocks();
 
+        assignmentSubmitter.submitCourseAttendances(true, attendanceSummaryModelList, COURSE_ID, oauthToken, assignmentConfigurationFromSetup);
+        verify(canvasApiWrapperService, times(1)).gradeMultipleSubmissionsByCourse(any(), any());
+    }
 
+    @Test
+    public void submitCourseAttendancesFailCanvasGradingError() throws Exception {
+        progressOptional.get().setWorkflowState("failed");
+        setupMocks();
+
+        try {
+            assignmentSubmitter.submitCourseAttendances(true, attendanceSummaryModelList, COURSE_ID, oauthToken, assignmentConfigurationFromSetup);
+            verify(canvasApiWrapperService, times(1)).gradeMultipleSubmissionsByCourse(any(), any());
+        } catch (AttendanceAssignmentException exception) {
+            Assert.assertEquals(AttendanceAssignmentException.Error.FAILED_PUSH, exception.error);
+        }
     }
 
     @Test
-    public void submitCourseAttendancesHappyPath() throws Exception {
+    public void deletedStudentsNotSubmitted() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00112)